lib/spack/spack/environment/environment.py

```diff
@@ -605,26 +638,18 @@ def regenerate(self, concretized_root_specs):
 
         view = self.view(new=new_root)
 
-        root_dirname = os.path.dirname(self.root)
-        tmp_symlink_name = os.path.join(root_dirname, "._view_link")
-
         # Create a new view
         try:
             fs.mkdirp(new_root)
             view.add_specs(*specs, with_dependencies=False)
-
-            # create symlink from tmp_symlink_name to new_root
-            if os.path.exists(tmp_symlink_name):
-                os.unlink(tmp_symlink_name)
-            symlink(new_root, tmp_symlink_name)
-
-            # mv symlink atomically over root symlink to old_root
-            fs.rename(tmp_symlink_name, self.root)
+            if use_renameat2:
+                spack.util.atomic_update.atomic_update_renameat2(new_root, self.root)
+            else:
+                spack.util.atomic_update.atomic_update_symlink(new_root, self.root)
```

Comment:
Why isn't this calling `atomic_update()` (which abstracts the check)?

Reply:
Because whether the ViewDescriptor uses renameat2 is a separate concern from whether it is available, and `atomic_update()` keys on whether it's available.